### PR TITLE
WIP: Fix test style in unused parentheses lint test

### PR DESCRIPTION
**Pull request by monoflo**
Hi! This is my first attempt at a pull request for the Rust project.

I'm fairly certain that I did this wrong, so I'm open to any and all feedback!

Per the issue, I was unable to find `test/lint/used_parens_remove_json_suggestion.rs`. Is this a file I should create? 

# Issue

#63237

### CODE REVIEW

**Comment by jonas-schievink** [comment]
> Per the issue, I was unable to find `test/lint/used_parens_remove_json_suggestion.rs`. Is this a file I should create?

I think the issue was just confusing the name of the file. According to a quick ripgrep, the other test using the `pretty-json` directive is `src/test/ui/lint/use_suggestion_json.rs`.

File: src/test/ui/lint/unused_parens_json_suggestion.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// compile-flags: --error-format pretty-json -Zunstable-options
+// compile-flags: --error-format -Zunstable-options
 // build-pass (FIXME(62277): could be check-pass?)
```
Comment (JohnTitor) [inline src/test/ui/lint/unused_parens_json_suggestion.rs]: This `build-pass` annotation should be removed

**Comment by monoflo** [comment]
From the Azure CI logs:

```
error: argument for --error-format must be `human`, `json` or `short` (instead was `-Zunstable-options`)
```

I set the argument to `json` per 6ebe5036409aa7aa93b182c081d8c6cf0e9fafbb. Is this correct?

**Comment by monoflo** [comment]
So I understand I have to edit the corresponding `.stderr` files in order for the test output to match up.

Could someone lend me a hand with this, or give me some pointers?

**Comment by nikomatsakis** [comment]
@monoflo usually the easiest way to get the `.stderr` files correct is to run `./x.py test --bless`, which will update them for you to match the compiler's current output =)

**Comment by JohnCSimon** [comment]
Ping from triage
@monoflo  moving this into the waiting on author pile as this has merge conflicts and there appears to be some work that has to be done.
Thank you for your contribution!

**Comment by JohnCSimon** [comment]
Pinging again from triage
@monoflo can you please resolve the conflicted file?
Thank you

**Comment by nikomatsakis** [comment]
(Note: you'll also want to run `--bless`)

**Comment by nikomatsakis** [comment]
@monoflo great, thanks! can you rebase to remove the merge commit?

**Comment by hdhoang** [comment]
ping from triage @monoflo, could you rebase so that only your commit appears in the "Commits" tab for this PR? Please feel free to reach out to us in #triage-wg on rust-lang Discord for more help.

File: src/test/ui/lint/unused_parens_json_suggestion.fixed [hunk 2]
```diff
@@ -1,5 +1,4 @@
-// compile-flags: --error-format pretty-json -Zunstable-options
-// build-pass (FIXME(62277): could be check-pass?)
+// compile-flags: --error-format json -Zunstable-options
```
Comment (nikomatsakis) [inline src/test/ui/lint/unused_parens_json_suggestion.fixed]: it seems like pretty-json is more "diff friendly", even so, no?

**Review by nikomatsakis — COMMENTED** [review]
Maybe we should keep up the "pretty-json" too?

cc @bravomikekilo, who opened the original issue (I don't have a strong opinion here)

**Comment by nikomatsakis** [comment]
Also: we are getting failures in the checks, are those maybe spurious? I couldn't get much from the logs.

**Comment by Dylan-DPC** [comment]
@monoflo any updates on this?

**Comment by hdhoang** [comment]
second ping from triage @monoflo, thank you for your work on this. When you can continue, please re-open this PR first then push your updates.

**Pull request closed without merging**